tests/backup/backup_helper.go

```diff
@@ -5996,3 +6004,59 @@ func CreateKubevirtBackupRuleForAllVMsInNamespace(ctx context.Context, namespace
 
 	return ruleStatus, ruleName, nil
 }
+
+func validateCRCleanup(resourceName string, clusterName string, orgID string,
+	resourceNamespace []string, ctx context.Context, crType string) error {
+	var allCRs []string
+	var err error
+
+	backupDriver := Inst().Backup
+	clusterUID, err := backupDriver.GetClusterUID(ctx, orgID, clusterName)
+	if err != nil {
+		return err
+	}
+
+	currentAdminNamespace, _ := getCurrentAdminNamespace()
+	if len(resourceNamespace) == 1 {
+		currentAdminNamespace = resourceNamespace[0]
+	}
+
+	driveName := Inst().Backup
+	clusterReq := &api.ClusterInspectRequest{OrgId: orgID, Name: clusterName, IncludeSecrets: true, Uid: clusterUID}
+	clusterResp, err := driveName.InspectCluster(ctx, clusterReq)
+	if err != nil {
+		return err
+	}
+	clusterObj := clusterResp.GetCluster()
+
+	validateCRCleanupInNamespace := func() (interface{}, bool, error) {
+		if crType == "restore" {
+			allCRs, err = GetRestoreCRs(currentAdminNamespace, clusterObj)
+		} else if crType == "backup" {
+			allCRs, err = GetBackupCRs(currentAdminNamespace, clusterObj)
+		} else {
+			return nil, true, fmt.Errorf("Please specify a valid option for CR validation. Options available - [backup, restore]")
+		}
+
+		if err != nil {
+			return nil, true, err
+		}
+
+		log.Infof("Validating CR cleanup")
+		log.InfoD("All CRs in [%s] are [%v]", currentAdminNamespace, allCRs)
+
+		for _, eachCR := range allCRs {
+			if strings.Contains(eachCR, resourceName) {
+				log.Infof("CR found for [%s] under [%s] namespace", allCRs, currentAdminNamespace)
+				return nil, true, fmt.Errorf("CR cleanup validation failed for - [%s]", resourceName)
+			}
+		}
+
+		return nil, false, nil
+	}
+
+	_, err = task.DoRetryWithTimeout(validateCRCleanupInNamespace, 5*time.Minute, 30*time.Second)
```

Comment:
Please make it 5 or 10 seconds instead of 30.

Reply:
Changed to 5 seconds.